src/ivoc/xmenu.cpp

```diff
@@ -1386,8 +1376,10 @@ void HocPanel::write(std::ostream& o) {
     //	o << "xpanel(\"" << getName() << "\")" << std::endl;
     Sprintf(buf, "xpanel(\"%s\", %d)", getName(), horizontal_);
     o << buf << std::endl;
-    for (i = 1; i < ilist_.count(); i++) {
-        ilist_.item(i)->write(o);
+    if (ilist_.size() > 1) {
```

Comment:
Why the additional `if`? Say if `ilist_size() == 1` then in the for loop `i < ilist_size()` fails and the body never executes.

Reply:
because the loop start at one

Reply:
the front element is a metadata storage

Reply:
Isn't
```
for(size_t i = 1; i < n; ++i) {
  std::cout << i << "\n";
}
```
equivalent to
```
if(i > 1) {
  for(size_t i = 1; i < n; ++i) {
    std::cout << i << "\n";
  }
}
```

For example if `n == 0` then the former will fail the very first check that: `i < n` and therefore print nothing. While the latter, will fail the `if` condition and also print nothing.

Reply:
I don't want to modify it :p